libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1301,7 +1332,13 @@ smt::Expression SMTEncoder::compoundAssignment(Assignment const& _assignment)
 
 void SMTEncoder::assignment(VariableDeclaration const& _variable, Expression const& _value)
 {
-	assignment(_variable, expr(_value, _variable.type()));
+	// In general, at this point, the SMT sorts of _variable and _value are the same,
+	// even if there is implicit conversion.
+	// This is a special case where the SMT sorts are different.
+	// For now we are unaware of other cases where this happens, but if they do appear
+	// we should extract this into an `implicitConversion` function.
+	if (_variable.type()->category() != Type::Category::Array || _value.annotation().type->category() != Type::Category::StringLiteral)
```

Comment:
So what happens in the ``else`` case?

Reply:
Then the constraint that says that the array/string/bytes has a certain value isn't added, and the array is just a fresh variable.
Maybe I add a TODO for the else case, where the solution would be to `smt::store` each string literal item in the array